src/AutoBuild.jl

```diff
@@ -160,8 +172,57 @@ function build_tarballs(ARGS, src_name, src_version, sources, script,
         error("Cannot register with a local deployment!")
     end
 
+    universe_path = nothing
+    if universe
+        if deploy
+            if deploy_repo != "universe"
+                error("When deploying from a universe, must use --deploy=universe")
+            end
+            deploy_universe = true
+            deploy_bin_repo = deploy_jll_repo = "JuliaBinaryWrappers/$(src_name)_jll.jl"
+        end
+
+        # Find the universe directory by ascending from pwd() until we find a "universes" directory
+        dir = pwd()
+        while !isdir(joinpath(dir, "universes"))
+            dir = dirname(dir)
+            if dir == "/"
+                error("Could not find `universes` diretory in any parent of pwd")
+            end
+        end
+        universe_path = joinpath(dir, "universes", universe_name)
+
+        if !isdir(universe_path)
+            if deploy_universe
+                mkpath(universe_path)
+
+                # Symlink the `packages`, `logs` and `artifacts` directories to the
+                # currently active repo
+                active_depot = first(Base.DEPOT_PATH)
+                for folder in ("packages", "logs", "artifacts")
+                    symlink(joinpath(active_depot, folder), joinpath(universe_path, folder))
+                end
+
+                # Create new `registries` and `clones` folders local to this registry
+                mkpath(joinpath(universe_path, "registries"))
+                mkpath(joinpath(universe_path, "clones"))
+
+                ctx = Pkg.Types.Context()
+                Pkg.GitTools.clone(ctx, "https://github.com/JuliaRegistries/General",
+                    joinpath(universe_path, "registries", "General"); header = "General registry to new universe")
+            else
+                @warn "Universe does not exist, but deploy not specified. Skipping Universe creation."
+                universe_path = nothing
+            end
+        end
+    end
+
     if deploy_bin || deploy_jll
-        code_dir = joinpath(Pkg.devdir(), "$(src_name)_jll")
+        if universe
+            code_dir = joinpath(universe_path, "clones", string(jll_uuid("$(src_name)_jll")))
```

Comment:
What's the reason for this?

Reply:
Makes the commit available to the regular logic to just check it out from `clones`

Reply:
Okay. Could alternatively put in the devdir path as the url in the registry, right? I guess this avoids one layer of indirection since Pkg would cache it here anyway.

Reply:
<img src="https://i.imgflip.com/4ds0uv.jpg" />

Yes, this seemed easier.

Reply:
I'm confused; the `clones` directory in my normal depot is a bunch of bare repositories, and the names are all numeric, not UUIDs.

Reply:
Yea, cold be stored wherever. I don't think Pkg will pick this up.